examples/ann/annoy_nsmlib_example.ipynb

```diff
@@ -0,0 +1,743 @@
+      "User 25\n",
+      "     Known positives:\n",
+      "        Toy Story (1995)\n",
+      "        Twelve Monkeys (1995)\n",
+      "        Dead Man Walking (1995)\n",
+      "     Recommended:\n",
+      "        Independence Day (ID4) (1996)\n",
+      "        Mission: Impossible (1996)\n",
+      "        Fargo (1996)\n",
+      "User 450\n",
+      "     Known positives:\n",
+      "        Kolya (1996)\n",
+      "        Devil's Own, The (1997)\n",
+      "        Contact (1997)\n",
+      "     Recommended:\n",
+      "        Hoodlum (1997)\n",
+      "        G.I. Jane (1997)\n",
+      "        Edge, The (1997)\n"
+     ]
+    }
+   ],
+   "source": [
+    "sample_recommendation_original(model, movielens, [3, 25, 450])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So pretty similar! How about speed?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 110,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "100 loops, best of 3: 4.19 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "sample_recommendation_original(model, movielens, [3, 25, 450], print_output=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 213,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "100 loops, best of 3: 2.72 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "sample_recommendation([3,25,450], model, movielens, print_output=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So a small difference at thes volumes, but let's see how it compares to an sklearn distance method as n increases:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 243,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def time_recommend_item_for_item():\n",
+    "    t = timeit.Timer(\"recommend_item_for_item(item_vectors[0].reshape(1, -1), data, 10)\")\n",
+    "    time = t.timeit(100)\n",
+    "    return time\n",
+    "\n",
+    "def time_Annoy_recommend_item_for_item():\n",
+    "    t = timeit.Timer(\"temp_t.get_nns_by_vector(item_vectors[0], 10)\")\n",
+    "    time = t.timeit(100)\n",
+    "    return time\n",
+    "\n",
+    "def time_nms_recommend_item_for_item():\n",
+    "    t = timeit.Timer(\"nms_idx.knnQuery(item_vectors[0])[0]\")\n",
+    "    time = t.timeit(100)\n",
+    "    return time\n",
+    "\n",
+    "Annoy_response_times = []\n",
+    "pairwise_response_times = []\n",
+    "nms_response_times = []\n",
+    "\n",
+    "f = item_vectors.shape[1]\n",
+    "\n",
+    "for i in range (1, 500, 50):\n",
+    "    data = np.repeat(item_vectors, i, axis=0)\n",
+    "    __builtin__.__dict__.update(locals())\n",
```

Comment:
I am not sure what is going here but it almost certainly shouldn't be going on :)